packages/driver/src/cy/commands/navigation.ts

```diff
@@ -496,6 +592,10 @@ export default (Commands, Cypress, cy, state, config) => {
     } catch (e) {} // eslint-disable-line no-empty
   })
 
+  Cypress.primaryOriginCommunicator.on('visit:url', ({ url }) => {
+    $utils.iframeSrc(Cypress.$autIframe, url)
```

Comment:
Are we losing any of the information or checks we have in place in the `changeIframeSrc` used for non-cross-origin navigation?

Reply:
`changeIframeSrc` is run in the spec bridge driver, which retains all that information, but it can't change the iframe src so it delegates that the primary driver. We only want to change the iframe src in the primary because `changeIframeSrc` would hit cross-origin errors as it attempts to interact with the window object.